hugo/content/posts/nix-bazel-crosspiling.md

```diff
@@ -0,0 +1,129 @@
+
+Unfortunately I couldn't figure out a way to configure C/C++ toolchains using Bazel's existing helper functions that I was happy with, so I ended up avoiding the Bazel "feature" system and configuring actions from scratch. While this lacks the dynamic configurability in the default Bazel C/C++ toolchains, I think it's a lot easier to follow and philosophically I try to keep as much out of Bazel CLI/`.bazelrc` flags as possible.
+
+C/C++ toolchains are configured using the helpers in [//bazel/toolchain_rules](https://github.com/ConsumingChaos/consumingchaos.github.io/tree/master/examples/nix-bazel-crosspiling/bazel/toolchain_rules). The main interface is the top level [llvm_cc_toolchain_config](https://github.com/ConsumingChaos/consumingchaos.github.io/tree/master/examples/nix-bazel-crosspiling/bazel/toolchain_rules/llvm_cc_toolchain_config.bzl) function. Based on the supplied `target` it delegates constructing the `action_config`s to the appropriate tool. For compilation, this is always [clang](https://github.com/ConsumingChaos/consumingchaos.github.io/tree/master/examples/nix-bazel-crosspiling/bazel/toolchain_rules/cc_tools/clang.bzl), and for archiving and stripping [llvm-ar](https://github.com/ConsumingChaos/consumingchaos.github.io/tree/master/examples/nix-bazel-crosspiling/bazel/toolchain_rules/cc_tools/llvm-ar.bzl) and [llvm-strip](https://github.com/ConsumingChaos/consumingchaos.github.io/tree/master/examples/nix-bazel-crosspiling/bazel/toolchain_rules/cc_tools/llvm-strip.bzl) respectively, but for linking it's split between the different linker "flavors":
+
+- [ld.lld](https://github.com/ConsumingChaos/consumingchaos.github.io/tree/master/examples/nix-bazel-crosspiling/bazel/toolchain_rules/cc_tools/ld.lld.bzl) for Linux/Android
+- [ld64.lld](https://github.com/ConsumingChaos/consumingchaos.github.io/tree/master/examples/nix-bazel-crosspiling/bazel/toolchain_rules/cc_tools/ld64.lld.bzl) for Apple
+- [lld-link](https://github.com/ConsumingChaos/consumingchaos.github.io/tree/master/examples/nix-bazel-crosspiling/bazel/toolchain_rules/cc_tools/lld-link.bzl) for Windows (MSVC)
+- [wasm-ld](https://github.com/ConsumingChaos/consumingchaos.github.io/tree/master/examples/nix-bazel-crosspiling/bazel/toolchain_rules/cc_tools/wasm-ld.bzl) for WASM
+
+The actual toolchain configurations are located under [//bazel/toolchains/cc](https://github.com/ConsumingChaos/consumingchaos.github.io/tree/master/examples/nix-bazel-crosspiling/bazel/toolchains/cc), grouped by platform.
+
+Rust toolchains, by comparison, are much easier to register. The Nix Flake provides a Rust toolchain with the standard library prebuilt per target platform, so the actual Bazel side configuration is minimal. The Rust toolchain configurations are located under [//bazel/toolchains/rust](https://github.com/ConsumingChaos/consumingchaos.github.io/tree/master/examples/nix-bazel-crosspiling/bazel/toolchains/rust), grouped by platform.
```

Comment:
Is a "Bazel side configuration" a thing? If not, just say, "Bazel configuration."